Sources/Run/SwiftPackageInfo.swift

```diff
@@ -128,14 +128,22 @@ extension ParsableCommand {
             swiftPackage.version = latestTag
         }
 
-        guard packageResponse.isProductValid else {
-            throw CleanExit.message(
-                """
-                Error: Invalid argument '--product <product>'
-                Usage: The product should match one of the declared products on \(swiftPackage.repositoryURL).
-                Found available products: \(packageResponse.availableProducts).
-                """
-            )
+        if packageResponse.isProductValid == false {
+            if let firstProduct = packageResponse.packageContent.products.first?.name,
+               swiftPackage.product == "Undefined" {
+                Console.default.lineBreakAndWrite("Invalid product: \(swiftPackage.product)")
+                Console.default.lineBreakAndWrite("Using first found product instead: \(firstProduct)")
+
+                swiftPackage.product = firstProduct
+            } else {
+                throw CleanExit.message(
+                    """
+                    Error: Invalid argument '--product <product>'
+                    Usage: The product should match one of the declared products on \(swiftPackage.repositoryURL).
+                    Found available products: \(packageResponse.availableProducts).
+                    """
+                )
```

Comment:
With added logic this will only happen when there are no products on the decoded package response (which is very unlikely to occur), I think we could update the error message a little bit, something like:

```suggestion
                throw CleanExit.message("Error: \(swiftPackage.repositoryURL) doesn't contain any product declared on Package.swift")
```

What do you think?

Reply:
My first thought was to use first package only if it not specified (product == "Undefined"), but now I think we can use this logic in any "invalid package" case, same as with versionTag.
Updated logic and message as you suggested.